### PR TITLE
File for TokenContext.Provider Created

### PR DESCRIPTION
**Pull request by aserputov**
Hey, @redxzeta. 
Fix : #200

### CODE REVIEW

File: src/context/TokenContext.js [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+  }
+
+  setAuthenticated(true);
+}, []);
 
 const TokenContext = React.createContext("");
 
 export default TokenContext;
+
+export function useAuth() {
```
Comment (redxzeta) [inline src/context/TokenContext.js:49]: can you rename to usePetAuth()

File: src/context/TokenContext.js [hunk 2]
```diff
@@ -1,5 +1,54 @@
-import React from "react";
+import React, { createContext, useContext, useEffect, useState } from "react";
+import jwtDecode from "jwt-decode";
+
+const initialState = { session: null, user: null, username: null };
```
Comment (redxzeta) [inline src/context/TokenContext.js:4]: init fields would be `{token:null, authenticated:false}`

File: src/context/TokenContext.js [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+export const AuthContext = createContext(initialState);
+
+const [token, setToken] = useState("");
+const [Authenticated, setAuthenticated] = useState(false);
```
Comment (redxzeta) [inline src/context/TokenContext.js:8]: we can combine this into one state

`const [state, setState] = useState(initialState);`

File: src/context/TokenContext.js [hunk 4]
```diff
@@ -1,5 +1,54 @@
...
+
+    const currentDate = new Date();
+    if (decodedToken.exp * 1000 < currentDate.getTime()) {
+      fetchFunction();
+    } else {
+      setToken(localStorage.getItem("token"));
+    }
+  }
+
+  setAuthenticated(true);
```
Comment (redxzeta) [inline src/context/TokenContext.js:42]: do the setState for authenticated

File: src/context/TokenContext.js [hunk 5]
```diff
@@ -1,5 +1,54 @@
...
+}, []);
 
 const TokenContext = React.createContext("");
 
 export default TokenContext;
+
+export function useAuth() {
+  const context = useContext(AuthContext);
+  if (context === undefined)
+    throw Error("useAuth must be used within AuthProvider");
```
Comment (redxzeta) [inline src/context/TokenContext.js:52]: usePetAuth

**Review by redxzeta — CHANGES_REQUESTED** [review]

File: src/context/TokenContext.js (hunk 5 above)
Comment (aserputov) [reply src/context/TokenContext.js]: ✅

File: src/context/TokenContext.js (hunk 1 above)
Comment (aserputov) [reply src/context/TokenContext.js]: ✅

File: src/context/TokenContext.js (hunk 2 above)
Comment (aserputov) [reply src/context/TokenContext.js]: ✅

File: src/context/TokenContext.js (hunk 3 above)
Comment (aserputov) [reply src/context/TokenContext.js]: ✅

File: src/context/TokenContext.js (hunk 4 above)
Comment (aserputov) [reply src/context/TokenContext.js]: ✅

**Review by redxzeta — APPROVED** [review]

**Pull request merged**